### PR TITLE
Open Rao integration

### PR DESCRIPTION
**Pull request by obrix**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: pypowsybl/rao/impl/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2023, RTE (http://www.rte-france.com)
```
Comment (HugoKulesza) [inline pypowsybl/rao/impl/__init__.py]: The date should be changed here

File: pypowsybl/rao/__init__.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2023, RTE (http://www.rte-france.com)
```
Comment (HugoKulesza) [inline pypowsybl/rao/__init__.py]: Same here

File: pypowsybl/rao/impl/util.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from pypowsybl import _pypowsybl
```
Comment (HugoKulesza) [inline pypowsybl/rao/impl/util.py:7]: The header is missing

File: pypowsybl/rao/impl/rao.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import io
```
Comment (HugoKulesza) [inline pypowsybl/rao/impl/rao.py:7]: Header also missing here

File: pypowsybl/_pypowsybl.pyi [hunk 5]
```diff
@@ -769,7 +784,7 @@ def get_extensions_names() -> List[str]: ...
 def get_extensions_information() -> SeriesArray: ...
 def create_security_analysis() -> JavaHandle: ...
 def create_sensitivity_analysis() -> JavaHandle: ...
-def save_network(network: JavaHandle, file: str, format: str, parameters: Dict[str,str], report: Optional[JavaHandle]) -> None: ...
+def save_network(network: JavaHandle, file: str, format: str, parameters: Dict[str,str], report:Optional [JavaHandle]) -> None: ...
```
Comment (HugoKulesza) [inline pypowsybl/_pypowsybl.pyi]: I think it should be cleaned

File: .github/workflows/dev-ci.yml [hunk 6]
```diff
@@ -99,6 +99,12 @@ jobs:
           cmake $GITHUB_WORKSPACE/cpp  -DBUILD_PYPOWSYBL_JAVA=OFF -DPYPOWSYBL_JAVA_LIBRARY_DIR=$GITHUB_WORKSPACE/dist/lib -DPYPOWSYBL_JAVA_INCLUDE_DIR=$GITHUB_WORKSPACE/dist/include
           cmake --build . --config Release
 
+      - name: Upload wheel
+        uses: actions/upload-artifact@v4
+        with:
+          name: pypowsybl-wheel-linux-${{ matrix.python.name }}
+          path: wheelhouse/*.whl
+
```
Comment (HugoKulesza) [inline .github/workflows/dev-ci.yml]: Should we keep this or was it just for debugging purpose ?

File: java/src/main/java/com/powsybl/python/rao/RaoCFunctions.java [hunk 7]
```diff
@@ -0,0 +1,160 @@
+package com.powsybl.python.rao;
```
Comment (HugoKulesza) [inline java/src/main/java/com/powsybl/python/rao/RaoCFunctions.java:8]: Header also missing here

File: cpp/pypowsybl-java/CMakeLists.txt
Comment (HugoKulesza) [inline cpp/pypowsybl-java/CMakeLists.txt:1]: I'm not sure to understand every change here
Do we add all the ortools libraries of the different platforms in the pom and then only link one here depending on the actual platform ?

File: cpp/pypowsybl-cpp/bindings.cpp [hunk 8]
```diff
@@ -16,6 +16,7 @@ class PYBIND11_EXPORT PyPowsyblError;
 }
 #include "powsybl-cpp.h"
 #include "pylogging.h"
+#include <iostream>
```
Comment (HugoKulesza) [inline cpp/pypowsybl-cpp/bindings.cpp]: I don't think it is used anywhere ?

File: cpp/pypowsybl-java/CMakeLists.txt
Comment (obrix) [reply cpp/pypowsybl-java/CMakeLists.txt:1]: Yes that's exactly what is done here

**Review by geofjamg — APPROVED** [review]

**Review by HugoKulesza — APPROVED** [review]

**Pull request merged**